app/src/main/java/nerd/tuxmobil/fahrplan/congress/favorites/StarredListFragment.kt

```diff
@@ -293,8 +293,8 @@ class StarredListFragment :
     }
 
     private fun deleteSession(session: Session) {
-        session.highlight = false
-        viewModel.delete(session)
+        val toBeDeleted = Session(session).apply { highlight = false }
+        viewModel.delete(toBeDeleted)
```

Comment:
Instead of creating a short-lived Session instance this could be `viewModel.removeHighlight(session)`. And then the view model could call `repository.deleteHighlight(session.sessionId)` (that has to be written).

Reply:
:heavy_check_mark: Refactored. I like it much better now, too.